src/matrix-utils.ts

```diff
@@ -322,6 +328,36 @@ export async function createRoom(
     },
   });
 
+  // Wait for the room to arrive
+  await new Promise<void>((resolve, reject) => {
+    const onRoom = async (room: Room) => {
+      if (room.roomId === (await createPromise).room_id) {
+        resolve();
+        cleanUp();
+      }
+    };
+    createPromise.catch((e) => {
+      reject(e);
+      cleanUp();
+    });
+
+    const cleanUp = () => {
+      client.off(ClientEvent.Room, onRoom);
+    };
+    client.on(ClientEvent.Room, onRoom);
+  });
+
+  const result = await createPromise;
+
+  console.log(`Creating ${ptt ? "PTT" : "video"} group call room`);
+
+  await client.createGroupCall(
+    result.room_id,
+    ptt ? GroupCallType.Voice : GroupCallType.Video,
+    ptt,
+    GroupCallIntent.Room
+  );
```

Comment:
I am very confused as to how this should be here or as to how did things ever work without it?

Reply:
What happened was: in https://github.com/vector-im/element-call/pull/471/ I moved group call creation to be the responsibility of `useLoadGroupCall`. I thought that the `createGroupCall` call here in `createRoom` was superfluous, so I removed it, and I also thought that `useLoadGroupCall` would still be able to choose between PTT and non-PTT correctly, since it takes a `createPtt` flag.

There were two problems with this: the `createPtt` flag doesn't necessarily indicate whether the user selected walkie-talkie mode in the dropdown, instead it just reflects whether the `?ptt` URL parameter is present. The second problem is that the code was still passing a PTT flag to `createRoom`, even though I removed that. That should've never happened - if we had finished migrating to TypeScript at the time of https://github.com/vector-im/element-call/pull/471/, the errors probably would've forced me to rethink my decisions.

Reply:
Ah, I see - thank you for the explanation!